scripts/plot_dispatch.py

```diff
@@ -35,6 +35,195 @@
+            df_demand=df_demand_time_filtered,
+            unit="W",
+            colors_odict=COLORS,
+        )
+
+        plt.grid()
+        plt.xlabel("Date (mm-dd)", loc="center", fontdict={"size": 17})
+        plt.ylabel("Power", loc="center", fontdict={"size": 17})
+        plt.xticks(fontsize=14)
+        plt.yticks(fontsize=14)
+        # format x-axis representing the dates
+        plt.gca().xaxis.set_major_formatter(mdates.DateFormatter("%Y-%m-%d"))
+        plt.gca().xaxis.set_major_locator(mdates.WeekdayLocator())
+
+        # Shrink current axis's height by 10% on the bottom
+        box = ax.get_position()
+        ax.set_position(
+            [box.x0, box.y0 + box.height * 0.15, box.width, box.height * 0.85]
+        )
+
+        # Simplify legend. As there is only one color per technology, there should
+        # be only one label per technology.
+        simple_labels_dict = {
+            "Battery": ["Battery out", "Battery in"],
+            "El. transmission external": ["El. import", "El. export"],
+            "El. transmission B-BB": [
+                "El. transmission in",
+                "El. transmission out",
+            ],
+            "El. shortage / curtailment": ["El. shortage", "Curtailment"],
+            "Heat cen. storage": ["Heat cen. storage out", "Heat cen. storage in"],
+            "Heat cen. mismatch": ["Heat cen. excess", "Heat cen. shortage"],
+            "Heat dec. storage": ["Heat dec. storage out", "Heat dec. storage in"],
+            "Heat dec. mismatch": ["Heat dec. excess", "Heat dec. shortage"],
+        }
+
+        handles, labels = reduce_labels(ax=ax, simple_labels_dict=simple_labels_dict)
+
+        # Put a legend below current axis
+
+        ax.legend(
+            handles=handles,
+            labels=labels,
+            loc="upper center",
+            bbox_to_anchor=(0.5, -0.25),
+            fancybox=True,
+            ncol=4,
+            fontsize=14,
+        )
+
+        # remove year from xticks
+        formatter = mdates.DateFormatter("%m-%d")
+        ax.xaxis.set_major_formatter(formatter)
+        locator = mdates.AutoDateLocator()
+        ax.xaxis.set_major_locator(locator)
+
+        fig.tight_layout()
+        file_name = (
+            bus_name + "_" + start_date[5:7] + config.settings.general.plot_filetype
+        )
+        plt.savefig(os.path.join(plotted, file_name), bbox_inches="tight")
+
+
+def get_df_for_aggregation(list_with_dfs):
+    """
+    This function adds DataFrames from a list in to a new Dataframe
+
+    Parameters
+    ----------
+    list_with_dfs: list
+        List containing DataFrames to be added to Dataframe `df_concatenated`
+
+    Returns
+    -------
+    df_concatenated: pd.DataFrame
+        Result DataFrame with concatenated DataFrames
+
+    """
+    df_concatenated = pd.DataFrame()
+
+    for df_in_list in list_with_dfs:
+        df_concatenated = df_concatenated.append(df_in_list, ignore_index=True)
+
```

Comment:
Could you do `df_concatenated = pd.concat(list_with_dfs)` instead?

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/242/commits/6a646904352794ae67c2588b0e951a7d40caff7a.